server/src/main/java/org/elasticsearch/TransportVersions.java

```diff
@@ -175,6 +175,7 @@ static TransportVersion def(int id) {
     public static final TransportVersion COHERE_BIT_EMBEDDING_TYPE_SUPPORT_ADDED_BACKPORT_8_X = def(8_840_0_01);
     public static final TransportVersion ELASTICSEARCH_9_0 = def(9_000_0_00);
     public static final TransportVersion COHERE_BIT_EMBEDDING_TYPE_SUPPORT_ADDED = def(9_001_0_00);
+    public static final TransportVersion ESQL_LOOKUP_JOIN_CONFIGURATION_IN_REQUEST = def(9_002_0_00);
```

Comment:
It can go back to 9.0 but not 8.x

Reply:
What do you mean here?

Reply:
Ah, sorry. That you can backport this to 9.0 but that you can't use this transport version to go back to 8.x. I don't know *precisely* how to backport transport changes to 8.x, but I believe  you need more transport versions for it.

Reply:
I did just read the [docs on versioning](https://github.com/elastic/elasticsearch/blob/77f686ed73687b591ea1ff38bcdd9900e6e5755d/docs/internal/Versioning.md#L141-L159), and added the 8.x and 9.0 transport versions for backporting

Reply:
I think, given "Backporting transport version changes to previous releases should only be done if absolutely necessary, as it is very easy to get wrong and break the release in a way that is very hard to recover from.", I'd not backport this at all. Just hit main and just have the one transport version. It doesn't feel worth rushing it given the chance of failure on the backports. And that the transport versions are a somewhat limited resource. And they can version conflict with each other which is a whole kettle of fun.

Reply:
Uhm I guess. It was as to avoid potential conflicts later with JOIN. But surely, this case may be a bit too much. Reverting that part